src/herder/HerderImpl.cpp

```diff
@@ -509,18 +509,46 @@ HerderImpl::processSCPQueue()
 void
 HerderImpl::processSCPQueueUpToIndex(uint64 slotIndex)
 {
-    while (true)
+    try
     {
-        SCPEnvelope env;
-        if (mPendingEnvelopes.pop(slotIndex, env))
-        {
-            getSCP().receiveEnvelope(env);
-        }
-        else
+        uint64 minSlotIndex = 0;
+        bool notTracking = !mHerderSCPDriver.trackingSCP();
```

Comment:
I would find it slightly easier if this was logically inverted (positive not negative) and named `herderWasTrackingSCP`. It's named in the present tense, but its purpose is to refer to the past state of the herder's tracking-scp-ness (state before current loop iteration). And it's tested in a context where it's less clear that the `&&` condition you're testing is an event-edge ("did the tracking-ness of the herder change") rather than a conjunction of two separate facts; it would be easier to read that fact off if it was spelled almost identically to the second conjunct in the condition.

Reply:
didn't think of the relationship to the method name, makes sense (I coded it up like this before)